src/java/org/apache/cassandra/db/lifecycle/Tracker.java

```diff
@@ -185,17 +185,12 @@ Throwable updateSizeTracking(Iterable<SSTableReader> oldSSTables, Iterable<SSTab
 
     public void addInitialSSTables(Iterable<SSTableReader> sstables)
     {
-        addInitialSSTablesWithoutUpdatingSize(sstables);
-        maybeFail(updateSizeTracking(emptySet(), sstables, null));
-        // no notifications or backup necessary
+        addSSTablesInternal(sstables, true, false, true);
     }
 
     public void addInitialSSTablesWithoutUpdatingSize(Iterable<SSTableReader> sstables)
     {
-        if (!isDummy())
-            setupOnline(sstables);
-        apply(updateLiveSet(emptySet(), sstables));
-        // no notifications or backup necessary
+        addSSTablesInternal(sstables, true, false, false);
```

Comment:
Yes, I mentioned those earlier in the ticket. 
My understanding is that they are not mixed with the regular SSTableAddedNotification thanks to the new flag "isInitialization" and memtable is set to null for them.

Reply:
In practice, they are mixed with the regular `SSTableAddedNotification`. The only place where the code choose to ignore those notifications based on the `isInitialization `flag is in `CompactionStrategyManager`. It seems that in `SSTablesGlobalTracker` we will process 2 notifications for each added SSTable.
I might be missing something as I do not understand the reason why those extra notification were added.

I believe that in 3.0 and 3.11 that change is a breaking one. If a fork relies on `INotificationConsumer` to plugin extra features, receiving some extra notification can break those features.

Reply:
Why 2 notifications per added sstable?
The way I see it is that SSTableGlobalTracker needs to be able to track all sstables, even those loaded on start (those marked with `isInitialization`). But this doesn't mean two times firing notification. Also the tests show it. The only place where we saw new notification is in `testDropSSTables` where we use 'tracker.addInitialSSTables' so we fire notification.

But I see your point with `INotificationConsumer`. I am not familiar how users exactly use it but I can imagine that change to add also notification for the initially loaded sstables might be unexpected. In that case, does it sound reasonable to create a completely new notification for those? 
/cc @pcmanus in case he as an author has anything to add here

Reply:
Sorry, I realized my confusion. For some reason I was convinced that we were already sending notifications during the initialization phase. Effectively the new notification is needed as we did not do it.